src/repr.rs

```diff
@@ -56,38 +58,46 @@ pub enum LoadType {
 }
 
 #[derive(Clone, Debug, PartialEq)]
-pub enum DomainExemption {
-    SubdomainMatch(String),
-    DomainMatch(String),
+pub struct DomainMatcher {
+    pub exact: Box<[String]>,
+    pub subdomain: Box<[String]>,
 }
 
-impl DomainExemption {
-    fn matches(&self, request: &Request) -> bool {
-        let domain = match *self {
-            DomainExemption::SubdomainMatch(ref domain) |
-            DomainExemption::DomainMatch(ref domain) => domain
+impl DomainMatcher {
+    fn matches(&self, url: &Url) -> bool {
+        let domain = match url.domain() {
+            Some(domain) => domain,
+            None => return false,
         };
-
-        if request.url.find(&format!("://{}", domain)).is_some() {
-            return true;
-        }
-        if let DomainExemption::SubdomainMatch(_) = *self {
-            if request.url.find(&format!(".{}", domain)).is_some() {
+        for candidate in &*self.exact {
+            if domain == candidate {
                 return true;
             }
         }
-
+        for suffix in &*self.subdomain {
+            match domain.len().cmp(&suffix.len()) {
+                Ordering::Equal if domain == suffix => return true,
```

Comment:
Let's add a test for a subdomain match against the main domain, since clearly that didn't used to work.